src/mmw/js/src/compare/views.js

```diff
@@ -1008,10 +1090,85 @@ function getTr55Tabs(scenarios) {
     ]);
 }
 
+function mapScenariosToHydrologyChartData(scenarios, key) {
+    return scenarios
+        .map(function(scenario) {
+            return scenario
+                .get('results')
+                .models
+                .reduce(function(accumulator, next) {
+                    if (next.get('displayName') !== models.constants.HYDROLOGY) {
+                        return accumulator;
+                    }
+
+                    return accumulator.concat(next
```

Comment:
Similarly to the above, can we use `map(...).reverse()` here instead of `.reduce()` with `accumulator.concat`?

Reply:
This `reduce` combines a `filter` and a `map` operation so changing it will involve another pass. In this case the output is not the same length as the input.

Reply:
Ah I see. Let's keep it then.